python/mxnet/gluon/data/dataset.py

```diff
@@ -185,6 +185,20 @@ def transform_first(self, fn, lazy=True):
         """
         return self.transform(_TransformFirstClosure(fn), lazy)
 
+    def flatten(self):
+        """Returns a new dataset with samples flattened.
+
+        It is usefull to call after transform() when your transformer function 'fn' transform
+        each sample into multiple samples.
+        Note that the items in dataset must be iterable, e.g., list.
+
+        Returns
+        -------
+        Dataset
+            The flattened dataset.
+        """
+        return SimpleDataset(list(itertools.chain.from_iterable(self)))
```

Comment:
would it make sense to include a lazy flatten option? Just a thought.

Reply:
I guess you mean having a new class for example, flattenedDataset which implements __getitem__ and __len__ ?

Reply:
@szha The overhead of triggering `__getitem__` and `__len__` instead of just flattening directly may be too large.

Reply:
OK